golem_messages/datastructures/tasks.py

```diff
@@ -55,7 +55,7 @@ class TaskHeader(datastructures.Container):
                 fail_msg="Subtask timeout is less than 0",
             ),
         ),
-        'task_type': (validators.validate_varchar128, ),
+        'market_type': (validators.validate_varchar128, ),
```

Comment:
I think it should validate enum. You could use logic similar to this one:
https://github.com/golemfactory/golem-messages/blob/051492623d99135ef6928abe9791062d8105406f/golem_messages/message/base.py#L296-L305

Reply:
What about using deserialize_market_type and trying to convert string to an Enum. If this string is not in the enum the deserialization will fail.

Reply:
1. `TaskHeader` is not a message. `TaskToCompute` is.
1. lno 299 does exactly that - tries to convert string/any value into enum.